src/main/java/qna/domain/aggregate/AnswerGroup.java

```diff
@@ -39,23 +40,13 @@ public boolean isEmpty() {
 		return answers.isEmpty();
 	}
 
-	public void validateIsSameWithUserAndWriter(User loginUser) throws CannotDeleteException {
-		for (Answer answer : answers) {
-			answer.validateIsOwner(loginUser);
-		}
+	public void checkAnswersAreSameWithUserAndWriter(User loginUser) {
+		answers.forEach(answer -> answer.checkIsOwner(loginUser));
 	}
 
-	public void deleteAll() {
-		answers.forEach(Answer::delete);
-	}
-
-	public List<DeleteHistory> generateDeleteHistoryAllOfAnswers() {
+	public List<DeleteHistory> deleteAll() {
```

Comment:
```suggestion
	public DeleteHistoryGroup deleteAll(User loginUser) {
```

1. 삭제하면서 권한 체크하도록 만들면 어떨까요
2. 일급 컬렉션을 활용해보는게 좋겠네요

Reply:
DM을 통해 말씀하신 내용을 이해했습니다 💡 💡 💡 
객체지향으로 사고하는 것이 정말 어려운 것이라는 것을 절실히 느꼈습니다 😢 😱 
말씀하신대로 반영하도록 하겠습니다!